### DISCUSSION TITLE
Should we let the user confirm/alter the models that the chatbot is going to use as context?

### DISCUSSION THREAD

**Discussion by codeblech** [pragunbhutani/dbt-llm-agent#40] [Ideas]
pinterest does so for their internal text-to-sql [tool](https://medium.com/pinterest-engineering/how-we-built-text-to-sql-at-pinterest-30bad30dabff)

**Comment by pragunbhutani** [comment]
Hmm this is an interesting approach 🤔 
So you're saying that before the bot comes up with an answer, it should confirm the models it has found to be relevant to the question?

On one hand, I like how this might prevent false answers but on the other, it will unnecessarily interrupt the workflow. Users can just as easily ask it to change the models it has used in a follow up query.

**Comment by codeblech** [comment]
Yeah, that was naive and would hinder the UX. At the risk of sounding totally out of touch with the project, the cursor approach might be good. The retrieved context can be displayed in chat UI alongside the actual output. Users can also add context using the `@` symbol to nudge the model/agent/retrieval in the right direction
